classes/task/completion.php

```diff
@@ -86,7 +86,20 @@ public function execute() {
             $totalview = (int)$record->get('totalview');
             $comments = (int)$record->get('comments');
             $replies = (int)$record->get('replies');
-
+            // Get an array of user IDs from completiondatarecords
```

Comment:
Instead of this logic do a reset to users completion state when completion settings change from internal moodle to annoto completion somewhat similar to reset_all_state():
```
$trackedusers = $this->get_tracked_users();
// Recalculate state for each kept user
foreach ($keepusers as $trackedusers) {
    $this->update_state($cm, COMPLETION_UNKNOWN, $keepuser);
}
```